src/riak_governor_ensemble_master.erl

```diff
@@ -48,11 +50,23 @@ handle_cast(ring_changed, #state{ringhash=RingHash}=State) ->
         RingHash ->
             {noreply, State};
         NewRingHash ->
-            ensure_ensembles_started(State),
+            case ensure_ensembles_started(State) of
+                ok -> ignore;
+                {error, Error} ->
+                    % schedule another ensemble start attempt
+                    error_logger:error_msg("Failed to create ensembles due to: ~p, will retry in ~bms", [Error, ?RETRY_DELAY]),
+                    timer:apply_after(?RETRY_DELAY, ?MODULE, ring_changed, [])
+            end,
             {noreply, State#state{ringhash=NewRingHash}}
     end;
 handle_cast(init_ensembles, State) ->
-    ensure_ensembles_started(State),
+    case ensure_ensembles_started(State) of
+        ok -> ignore;
+        {error, Error} ->
+            % schedule another ensemble start attempt
+            error_logger:error_msg("Failed to create ensembles due to: ~p, will retry in ~bms", [Error, ?RETRY_DELAY]),
+            timer:apply_after(?RETRY_DELAY, ?MODULE, ring_changed, [])
+    end,
```

Comment:
"really bad practice" is an opinion I really badly disagree with, but will comply anyway :)